client/src/components/language-switcher/index.tsx

```diff
@@ -21,11 +27,19 @@ export const LanguageSwitcher = ({
   });
 
   const handleClick = () => {
-    setActiveLanguage(
-      activeLanguage === ActiveLanguage.EN
-        ? ActiveLanguage.UA
-        : ActiveLanguage.EN,
-    );
+    if (activeLanguage === ActiveLanguage.EN) {
```

Comment:
I don't see any reason to duplicate that logic for each of the languages. You can do it simpler:

    const newLanguage= activeLanguage === ActiveLanguage.EN 
      ? ActiveLanguage.UA 
      : ActiveLanguage.EN;

    setActiveLanguage(newLanguage);

    const path = pathname.replace(local, newLanguage);

    router.replace(path);